File: javapatterns/src/main/java/org/iut/nantes/state/canards/etats/SurTerre.java
package org.iut.nantes.state.canards.etats;

import org.iut.nantes.state.canards.Canard;

public class SurTerre implements Etat {
    private Canard canard;

    public SurTerre (Canard canard) {
        this.canard = canard;
    }

    @Override
    public void avancer() {
        System.out.println("Je marche.");
    }

    @Override
    public void plongeon() {

    }

    @Override
    public void trempette() {
        canard.changeEtatSurLeau();
    }

    @Override
    public void decollage() {
        canard.changeEtatDansLair();
    }

    @Override
    public void palmeATerre() {

    }
}

File: kotlinpatterns/src/main/kotlin/org/iut/nantes/state/canards/etats/SurTerre.kt
package org.iut.nantes.state.canards.etats

import org.iut.nantes.state.canards.Canard

class SurTerre(private val canard: Canard) : Etat {
    override fun avancer() {
        println("Je marche sur le sol")
    }

    override fun plongeon(){ }

    override fun trempette() {
        canard.changeEtatSurEau()
    }

    override fun decollage() {
        canard.changeEtatDansAir()
    }

    override fun palmeATerre() {
        println("Déjà sur terre!")
    }
}